src/vm/amd64/umthunkstub.S

```diff
@@ -0,0 +1,567 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Copyright (c) Geoff Norton. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+// ==++==
+//
+
+//
+// ==--==
+
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+#ifdef FEATURE_MIXEDMODE
+#define IJWNOADThunk__MakeCall        ?MakeCall@IJWNOADThunk@@KAXXZ 
+#define IJWNOADThunk__FindThunkTarget ?FindThunkTarget@IJWNOADThunk@@QEAAPEBXXZ
+#endif
+#define gfHostConfig                  ?g_fHostConfig@@3KA
+#define NDirect__IsHostHookEnabled    ?IsHostHookEnabled@NDirect@@SAHXZ
+
+//
+// METHODDESC_REGISTER: UMEntryThunk*
+//
+NESTED_ENTRY TheUMEntryPrestub, _TEXT, UMEntryPrestubUnwindFrameChainHandler
+
+#define TheUMEntryPrestub_XMM_SAVE_OFFSET SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES
+
+// Ensure that the new rsp will be 16-byte aligned.  Note that the caller has
+// already pushed the return address.
+#if (((SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES + SIZEOF_MAX_FP_ARG_SPILL) + 8) % 16) != 0
+#define TheUMEntryPrestub_STACK_FRAME_SIZE (SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES + SIZEOF_MAX_FP_ARG_SPILL + 8)
+#else
+#define TheUMEntryPrestub_STACK_FRAME_SIZE (SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES + SIZEOF_MAX_FP_ARG_SPILL)
+#endif
+
+        alloc_stack     TheUMEntryPrestub_STACK_FRAME_SIZE
+
+        save_reg_postrsp    rdi, TheUMEntryPrestub_STACK_FRAME_SIZE + 0x8
+        save_reg_postrsp    rsi, TheUMEntryPrestub_STACK_FRAME_SIZE + 0x10
+        save_reg_postrsp    rdx,  TheUMEntryPrestub_STACK_FRAME_SIZE + 0x18
+        save_reg_postrsp    rcx,  TheUMEntryPrestub_STACK_FRAME_SIZE + 0x20
+
+        save_xmm128_postrsp xmm0, TheUMEntryPrestub_XMM_SAVE_OFFSET
+        save_xmm128_postrsp xmm1, TheUMEntryPrestub_XMM_SAVE_OFFSET + 0x10
+        save_xmm128_postrsp xmm2, TheUMEntryPrestub_XMM_SAVE_OFFSET + 0x20
+        save_xmm128_postrsp xmm3, TheUMEntryPrestub_XMM_SAVE_OFFSET + 0x30
```

Comment:
and here we need to save xmm4-xmm7 on Linux/Mac, right? Based on this, the frame size calculated in this file is incorrect because it is using constants with Windows specific values (SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES, SIZEOF_MAX_FP_ARG_SPILL are currently defined for Windows only)